contracts/pools/weighted/WeightedPool2Tokens.sol

```diff
@@ -0,0 +1,1063 @@
+    // These factors are always greater than or equal to one: tokens with more than 18 decimals are not supported.
+    uint256 internal immutable _scalingFactor0;
+    uint256 internal immutable _scalingFactor1;
+
+    event SwapFeePercentageChanged(uint256 swapFeePercentage);
+
+    modifier onlyVault(bytes32 poolId) {
+        _require(msg.sender == address(getVault()), Errors.CALLER_NOT_VAULT);
+        _require(poolId == getPoolId(), Errors.INVALID_POOL_ID);
+        _;
+    }
+
+    constructor(
+        IVault vault,
+        string memory name,
+        string memory symbol,
+        IERC20 token0,
+        IERC20 token1,
+        uint256 normalizedWeight0,
+        uint256 normalizedWeight1,
+        uint256 swapFeePercentage,
+        uint256 pauseWindowDuration,
+        uint256 bufferPeriodDuration,
+        address owner
+    )
+        // Base Pools are expected to be deployed using factories. By using the factory address as the action
+        // disambiguator, we make all Pools deployed by the same factory share action identifiers. This allows for
+        // simpler management of permissions (such as being able to manage granting the 'set fee percentage' action in
+        // any Pool created by the same factory), while still making action identifiers unique among different factories
+        // if the selectors match, preventing accidental errors.
+        Authentication(bytes32(uint256(msg.sender)))
+        BalancerPoolToken(name, symbol)
+        BasePoolAuthorization(owner)
+        TemporarilyPausable(pauseWindowDuration, bufferPeriodDuration)
+    {
+        _setSwapFeePercentage(swapFeePercentage);
+
+        bytes32 poolId = vault.registerPool(IVault.PoolSpecialization.TWO_TOKEN);
+
+        // Pass in zero addresses for Asset Managers
+        IERC20[] memory tokens = new IERC20[](2);
+        tokens[0] = token0;
+        tokens[1] = token1;
+        vault.registerTokens(poolId, tokens, new address[](2));
+
+        // Set immutable state variables - these cannot be read from during construction
+        _vault = vault;
+        _poolId = poolId;
+
+        _token0 = token0;
+        _token1 = token1;
+
+        _scalingFactor0 = _computeScalingFactor(token0);
+        _scalingFactor1 = _computeScalingFactor(token1);
+
+        // Ensure each normalized weight is above them minimum and find the token index of the maximum weight
+        _require(normalizedWeight0 >= _MIN_WEIGHT, Errors.MIN_WEIGHT);
+        _require(normalizedWeight1 >= _MIN_WEIGHT, Errors.MIN_WEIGHT);
+
+        // Ensure that the normalized weights sum to ONE
+        uint256 normalizedSum = normalizedWeight0.add(normalizedWeight1);
+        _require(normalizedSum == FixedPoint.ONE, Errors.NORMALIZED_WEIGHT_INVARIANT);
+
+        _normalizedWeight0 = normalizedWeight0;
+        _normalizedWeight1 = normalizedWeight1;
+        _maxWeightTokenIndex = normalizedWeight0 > normalizedWeight1 ? 0 : 1;
```

Comment:
```suggestion
        _maxWeightTokenIndex = normalizedWeight0 >= normalizedWeight1 ? 0 : 1;
```

Extremely minor, but it seems slightly more intuitive that a 50/50 Pool would pay fees in the first token, not the second one.